### PR TITLE
Clarifying cookie options and session upgrade path

### PR DESCRIPTION
**Pull request by robertjd**

### CODE REVIEW

File: docs/upgrading.rst [hunk 1]
```diff
@@ -237,13 +237,22 @@ The above also applies to all other expansion options.
...
 Session management has also changed.  We now issue OAuth access tokens and
 refresh tokens when a user logs in with a username and password.  These are
-stored in the browser in HTTP-Only, Secure cookies.  This means that we no
-longer need a local cookie session mechanism, as the session is managed by
-Stormpath via these tokens.  Thus the ``secretKey`` option has been deprecated,
-and the timeout settings for cookies are now tied to the TTL settings of the
-Access Tokens and Refresh Tokens.  These can be modified on the OAuth Policy
-of your Stormpath Application, which is easily accessible through your
-Stormpath Admin Console.
+stored in the browser in HTTP-Only, Secure cookies.
```
Comment (typerandom) [inline docs/upgrading.rst]: "HTTP-Only, Secure cookies" -> "HTTP-only secure cookies"?

File: docs/upgrading.rst [hunk 2]
```diff
@@ -237,13 +237,22 @@ The above also applies to all other expansion options.
...
+
+This means that we no use a session middleware, as these token are managed by
```
Comment (typerandom) [inline docs/upgrading.rst]: "we no use a" -> "we no longer use a".

File: docs/authentication.rst [hunk 3]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
 please login to the Stormpath Admin Console and navigate to the OAuth policy of
 your Stormpath Application.  There you will find the settings for each token.
 
+.. _configuring_cookie_flags:
+
+Configuring Cookie Flags
+........................
+
+This library creates two cookies, one for the access token and one for the
```
Comment (typerandom) [inline docs/authentication.rst]: "access token" -> "Access Token"?

File: docs/authentication.rst [hunk 4]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
...
+refresh token.  There are several options available for controlling the flags
```
Comment (typerandom) [inline docs/authentication.rst]: "refresh token" -> "Refresh Token"?

File: docs/authentication.rst [hunk 5]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
...
+
++-------------+---------+------------------------------------------------------+
+| Cookie Flag | Default | Description                                          |
++=============+=========+======================================================+
+| domain      | null    | Set if needed, e.g. "subdomain.mydomain.com"         |
++-------------+---------+------------------------------------------------------+
+| httpOnly    | true    | True by default, do not disable without good reason  |
+|             |         | (exposes tokens to XSS | attacks).                   |
++-------------+---------+------------------------------------------------------+
+| path        | "/"     | Set if needed, e.g. "/newapp"                        |
```
Comment (typerandom) [inline docs/authentication.rst]: No terminal punctuation in description.

File: docs/authentication.rst [hunk 6]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
...
++-------------+---------+------------------------------------------------------+
+| secure      | null    | Will be ``true`` in HTTPS environments (as detected  |
```
Comment (typerandom) [inline docs/authentication.rst]: No terminal punctuation in description.

File: docs/authentication.rst [hunk 7]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
...
+      }
+    }
+  }
+
+This table describes each setting in detail:
+
++-------------+---------+------------------------------------------------------+
+| Cookie Flag | Default | Description                                          |
++=============+=========+======================================================+
+| domain      | null    | Set if needed, e.g. "subdomain.mydomain.com"         |
```
Comment (typerandom) [inline docs/authentication.rst]: No terminal punctuation in description.

File: docs/authentication.rst [hunk 8]
```diff
@@ -57,6 +58,50 @@ If you need to change the expiration time of the Access Token or Refresh Token,
...
++-------------+---------+------------------------------------------------------+
+| httpOnly    | true    | True by default, do not disable without good reason  |
+|             |         | (exposes tokens to XSS | attacks).                   |
++-------------+---------+------------------------------------------------------+
+| path        | "/"     | Set if needed, e.g. "/newapp"                        |
++-------------+---------+------------------------------------------------------+
+| secure      | null    | Will be ``true`` in HTTPS environments (as detected  |
+|             |         | by ``req.protocol``), unless explicitly set to       |
+|             |         | ``false`` (not recommended!)                         |
```
Comment (typerandom) [inline docs/authentication.rst]: No terminal punctuation in description.

File: docs/upgrading.rst [hunk 9]
```diff
@@ -237,13 +237,22 @@ The above also applies to all other expansion options.
...
+the  Stormpath API.  Thus, the following has changed:
+
+- The property ``req.stormpathSession`` has been removed, if you were using this
+  property to store stateful session information you will need to add a session
+  middleware to your library, such as `express-session`_.
```
Comment (typerandom) [inline docs/upgrading.rst]: "_." -> "."?

**Comment by typerandom** [comment]
Added some comments. But besides this, I think it looks good! :+1: :)

**Comment by robertjd** [comment]
Thanks @typerandom for the review!

**Pull request merged**